hastexo-guacamole-client/guacamole/client.py

```diff
@@ -0,0 +1,226 @@
+"""
+The MIT License (MIT)
+
+Copyright (c)   2014 rescale
+                2014 - 2016 Mohab Usama
+"""
+
+import socket
+import logging
+
+from .exceptions import GuacamoleError
+
+from .instruction import INST_TERM
+from .instruction import GuacamoleInstruction as Instruction
+
+# supported protocols
+PROTOCOLS = ('vnc', 'rdp', 'ssh')
+
+PROTOCOL_NAME = 'guacamole'
+
+BUF_LEN = 4096
+
+guac_logger = logging.getLogger(__name__)
+guac_logger.setLevel(logging.INFO)
+guac_logger.handlers = [logging.StreamHandler()]
+
+
+class GuacamoleClient(object):
+    """Guacamole Client class."""
+
+    def __init__(self, host, port, timeout=20, debug=True, logger=None):
+        """
+        Guacamole Client class. This class can handle communication with guacd
+        server.
+
+        :param host: guacd server host.
+
+        :param port: guacd server port.
+
+        :param timeout: socket connection timeout.
+
+        :param debug: if True, default logger will switch to Debug level.
+        """
+        self.host = host
+        self.port = port
+        self.timeout = timeout
+
+        self._client = None
+
+        # handshake established?
+        self.connected = False
+
+        # Receiving buffer
+        self._buffer = bytearray()
+
+        # Client ID
+        self._id = None
+
+        self.logger = guac_logger
+        if logger:
+            self.logger = logger
+
+        if debug:
+            self.logger.setLevel(logging.DEBUG)
+
+    @property
+    def client(self):
+        """
+        Socket connection.
+        """
+        if not self._client:
+            self._client = socket.create_connection(
+                (self.host, self.port), self.timeout)
+            self.logger.debug('Client connected with guacd server (%s, %s, %s)'
+                              % (self.host, self.port, self.timeout))
+
+        return self._client
+
+    @property
+    def id(self):
+        """Return client id"""
+        return self._id
+
+    def close(self):
+        """
+        Terminate connection with Guacamole guacd server.
+        """
+        self.client.close()
+        self._client = None
+        self.connected = False
+        print('Connection closed.')
```

Comment:
This looks like it belongs in a `self.logger.<level>` call rather than `print()`.

Reply:
It looks like this `print()` has been deleted; at least I can't find it anymore. :)

Reply:
replaced with `self.logger.info('Connection closed.')`